### PR TITLE
Allow running as non-root user for Rancher and Portainer

### LINKED ISSUE #33: Add PUID and PGID env option for Portainer compatibility
Heya,

I've seen that someone else already created a similar issue found here: https://github.com/Phyremaster/papermc-docker/issues/5

However this doesn't really work well with Portainer since it always runs the container as root no matter what, and you can't change this easily without changing your whole Docker installation. This makes updating plugin configs annoying since they always get saved as "root" then which you first have to chown again before modifying them properly.

Please add such an option, most other Docker containers support such an option specifically to avoid conflicts like these :)

Thanks in advance!

### PR DESCRIPTION
**Pull request by CourteousCoder**
Resolves #33 and improves quality of life when using orchestration tools.

### CODE REVIEW

**Comment by CourteousCoder** [comment]
I have these changes on dockerhub as `courteouscoder/papermc:latest` if you would like to try it out.
(Edited, after a username change.Not that this is relevant anymore)

**Comment by DanielSchetritt** [comment]
Sure thing!  The reason I use `tmux` is because the server process has already been started by the dockerfile `CMD` directive. So the only way to interact with the server is to have access to that TTY for the docker `CMD` However, in Rancher, there's not an easy way (to my knowledge) to hook into it since it's being orchestrated by kubernetes. Even things like the container name can change on reboot. But I do have access to a bash prompt, so the easiest way to run commands into the paper server console is to either use rcon (which appears to be not supported by this project) or to drop into a shell and attach the `tmux` session running the server.

I think this solution works well for both orchestration and plain-old docker because you don't _have_ to attach a new client to the tmux session. You could just use the one and it should function as expected.

If you don't like tmux, it can be changed to GNU screen or any other alternative. You just might have tweak it a bit.

I noticed some log viewers for docker/kubernetes will show the HUD of tmux in the logs, which looks a little unclean. I am sure it can be tweaked to make it cleaner. I'll look into that later, but for now I think it's still good to go, with your testing and approval.

**Comment by DanielSchetritt** [comment]
TL;DR The purpose of this PR is for compatibility with things like Rancher or Portainer. Using tmux makes it a bit easier to work with for these tools

**Comment by DanielSchetritt** [comment]
I just found a way to disable the tmux status bar. I'll add that in to make the logs cleaner.

**Comment by Phyremaster** [comment]
Oh crap, I meant to test this over the weekend and totally forgot. My bad. I'll do it within the next few days.

`tmux` being added is fine so long as `docker attach` still works as expected (I'll test that myself).

As for RCON, it's not "unsupported" and should still work if the port is published when the container is created/started. If this is not the case (i.e. if Docker requires an `EXPOSE` directive for a port to be published) we can reopen that issue and fix it.

**Comment by DanielSchetritt** [comment]
I had trouble with rcon, but I'll have to confirm whether it's an issue on my end or in the container. As for tmux, I just noticed a typo in my last commit 940ba08 . I am going to amend the commit, so you might have to delete your local copy of my branch before you pull.

**Review by Phyremaster — CHANGES_REQUESTED** [review]
Apologies for not responding for years. See #74 for what happened.

If Rancher and/or Portainer still haven't updated to use Docker's official method for setting the container's user, feel free to update this and reopen the PR, but I'm closing it for now.
